lib/cn/cn_tree.c

```diff
@@ -2354,6 +2356,84 @@ cn_tree_fini(void)
     }
 }
 
+static int
+kblk_compare(const void *a, const void *b)
+{
+    const struct kvset_kblk *kblk_a = a, *kblk_b = b;
+
+    /* Return the kblock with the largest min key. */
+    return -1 * key_disc_cmp(&kblk_a->kb_kdisc_min, &kblk_b->kb_kdisc_min);
+}
+
+static merr_t
+get_split_key(const struct cn_tree_node *node, void *key_buf, size_t key_buf_sz, uint16_t *key_len)
```

Comment:
I might put this function in a new file `split_key.c`, move the other_iter code into that file, and make everything about other iter private to `split_key.c`

Reply:
All the stuff existing in the same file seems pretty reasonable. I think it would help file organization more if our folders weren't so flat. For instance, this could go in `cn/node/split.c` or something like that. Not that I am advocating for that in this PR, but something to think about.

`cn/kvset/...`...

Reply:
I like to pass in the IN params first, and the OUT params last, plus should be more consistent with most of our code..